Chocobozzz/PeerTube
# User dropdown and notifications popover improvements

## Description

- Move the user dropdown on the avatar / username and extract the notifications popover to the right ;
**Looks more common and the old style was hard to understand for some people.**
- Improve the notifications popover appearance ;

<div style="text-align:center; display: flex; align-items: top">
<img src=https://user-images.githubusercontent.com/1877318/99911126-8f21d900-2cf2-11eb-95d6-23ab0eeffcd3.png />
<br />
<img src=https://user-images.githubusercontent.com/1877318/99911120-8d581580-2cf2-11eb-8387-7b72de6e00d0.png />
<img src=https://user-images.githubusercontent.com/1877318/99911124-8e894280-2cf2-11eb-8848-0651e3f2cebd.png />
</div>

- On mobile view (under 500px width), it now uses a link to /my-account/notifications instead of displaying the popover ;

- Add scrollToTop on all active links of the user dropdown (Profile, ..) and the notification popover.


## Related issues

- Fix the dropdown and popover detaching or latency jump for re-positioning effect when scrolling window or menu ;
- Fix sensitive icon hover top in user dropdown.

## Has this been tested?

- [x] 👍 yes, light tests as follows are enough

Tested on desktop view for the user dropdown and the notifications popover : 
- Make sure scrolling body and menu doesn't detach when opened ;
- Make sure the element is closed after clicking on a inside link or everywhere outside ;
- Make sure the user dropdown disapears when body scroll or and stays fixed when menu scroll ;
- Make sure the notifications popover stays fixed when body or menu scroll ;
- Make sure the active effect of the user avatar and name, and the notifications icon are well synced with dropdown or popover events ;

Tested on mobile view :
- Make sure the notifications popover doesn't display, instead a link to /my-account/notifications ;
- Make sure the user dropdown fits well the screen width and is attached to the menu when scroll. 


## Demos

On desktop : 
![user-menu-demo-desktop](https://user-images.githubusercontent.com/1877318/99911270-78c84d00-2cf3-11eb-954b-c1d05164ae64.gif)

On mobile : 
![user-menu-demo-mobile](https://user-images.githubusercontent.com/1877318/99911272-7ebe2e00-2cf3-11eb-98fc-eb901e00a51d.gif)
